src/main/webapp/app/exercises/programming/manage/services/programming-exercise.service.ts

```diff
@@ -35,6 +35,17 @@ export class ProgrammingExerciseService {
             .pipe(map((res: EntityResponseType) => this.convertDateFromServer(res)));
     }
 
+    /**
+     * Triggers the creation and setup of a new programming exercise
+     * @param programmingExercise
+     */
+    automaticSetupWithoutLocalSetup(programmingExercise: ProgrammingExercise): Observable<EntityResponseType> {
```

Comment:
again, I would separate this into a different service

Reply:
I created a new service and moved the method declaration there